crates/nu-command/src/platform/ansi/gradient.rs

```diff
@@ -0,0 +1,319 @@
+                    let fg_gradient = Gradient::new(fg_start, fg_end);
+                    let bg_gradient = Gradient::new(bg_start, bg_end);
+                    let gradient_string = build_all_gradient_text(val, fg_gradient, bg_gradient);
+                    Value::string(gradient_string, *span)
+                }
+                (Some(fg_start), None, Some(bg_start), Some(bg_end)) => {
+                    // Error - missing fg_end, so assume black
+                    let fg_end = Rgb::new(0, 0, 0);
+                    let fg_gradient = Gradient::new(fg_start, fg_end);
+                    let bg_gradient = Gradient::new(bg_start, bg_end);
+                    let gradient_string = build_all_gradient_text(val, fg_gradient, bg_gradient);
+                    Value::string(gradient_string, *span)
+                }
+                (Some(fg_start), Some(fg_end), None, None) => {
+                    // Foreground Only
+                    let gradient = Gradient::new(fg_start, fg_end);
+                    let gradient_string = gradient.build(val, TargetGround::Foreground);
+                    Value::string(gradient_string, *span)
+                }
+                (Some(fg_start), Some(fg_end), None, Some(bg_end)) => {
+                    // Error - missing bg_start, so assume black
+                    let bg_start = Rgb::new(0, 0, 0);
+                    let fg_gradient = Gradient::new(fg_start, fg_end);
+                    let bg_gradient = Gradient::new(bg_start, bg_end);
+                    let gradient_string = build_all_gradient_text(val, fg_gradient, bg_gradient);
+                    Value::string(gradient_string, *span)
+                }
+                (Some(fg_start), Some(fg_end), Some(bg_start), None) => {
+                    // Error - missing bg_end, so assume black
+                    let bg_end = Rgb::new(0, 0, 0);
+                    let fg_gradient = Gradient::new(fg_start, fg_end);
+                    let bg_gradient = Gradient::new(bg_start, bg_end);
+                    let gradient_string = build_all_gradient_text(val, fg_gradient, bg_gradient);
+                    Value::string(gradient_string, *span)
+                }
+                (Some(fg_start), Some(fg_end), Some(bg_start), Some(bg_end)) => {
+                    // Foreground and Background Gradient
+                    let fg_gradient = Gradient::new(fg_start, fg_end);
+                    let bg_gradient = Gradient::new(bg_start, bg_end);
+                    let gradient_string = build_all_gradient_text(val, fg_gradient, bg_gradient);
+                    Value::string(gradient_string, *span)
+                }
+            }
+        }
+        other => {
+            let got = format!("value is {}, not string", other.get_type().to_string());
+
+            Value::Error {
+                error: ShellError::TypeMismatch(
+                    got,
+                    other.span().unwrap_or(Span { start: 0, end: 0 }),
```

Comment:
You might want to `unwrap_or(*command_span)` instead. Pointing at span 0,0 will bring up a potentially random source file.